Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -266,27 +266,71 @@ public class GraphTraverser: GraphTraversing {
             test: isDependencyPrecompiledDynamicAndLinkable,
             skip: canDependencyEmbedProducts
         )
-        // Skip merged precompiled libraries from merging into the runnable binary
-        if case let .manual(dependenciesToMerge) = target.target.mergedBinaryType {
-            precompiledFrameworks = precompiledFrameworks
-                .filter { !isXCFrameworkMerged(dependency: $0, expectedMergedBinaries: dependenciesToMerge) }
-        }
-        references.formUnion(precompiledFrameworks.lazy.compactMap { self.dependencyReference(
-            to: $0,
-            from: .target(name: name, path: path)
-        ) })
-
         /// Other targets' frameworks.
         var otherTargetFrameworks = filterDependencies(
             from: .target(name: name, path: path),
             test: isDependencyDynamicTarget,
             skip: canDependencyEmbedProducts
         )
 
+        // Skip merged precompiled libraries from merging into the runnable binary first
```

Comment:
If the binary is merged and the framework is mergeable, for Debug, it will be reexported, while for the Release, it will become a static framework, being a part of this merged binary.

So adding them into the embedded frameworks list makes no sense.